src/PhpcsDiff.php

```diff
@@ -55,18 +57,43 @@ public function __construct(array $argv, CLImate $climate)
             $this->isVerbose = true;
         }
 
-        if (!isset($this->argv[1])) {
-            $this->error('Please provide a <bold>base branch</bold> as the first argument.');
-            return;
+        $this->ruleset = $this->getOption('--ruleset', self::DEFAULT_RULESET);
+
+        $this->baseBranch = $this->getArgument(0, '');
+        $this->currentBranch = $this->getArgument(1, '');
+
+        if ($this->isVerbose) {
+            if (!$this->baseBranch && !$this->currentBranch) {
+                $this->climate->comment("Comparing unstaged changes.");
+            } else {
+                $this->climate->comment(
+                    'Comparing branch: "' . $this->baseBranch . '" against: "' . $this->currentBranch . '"'
+                );
+            }
         }
+    }
 
-        $this->baseBranch = 'origin/' . str_replace('origin/', '', $this->argv[1]);
-        $this->currentBranch = trim(shell_exec('git rev-parse --verify HEAD'));
+    protected function getArgument($index, $default = null)
+    {
+        $arguments = array_filter($this->argv, function ($val) {
+            return strpos($val, '-') === false;
+        });
+        $arguments = array_values($arguments);
 
-        if (empty($this->currentBranch)) {
-            $this->error('Unable to get <bold>current</bold> branch.');
-            return;
+        return isset($arguments[$index]) ? $arguments[$index] : $default;
+    }
+
+    protected function getOption($name, $default = null)
+    {
+        foreach ($this->argv as $arg) {
+            list($flag, $val) = explode('=', $arg);
+
+            if ($flag == $name) {
```

Comment:
Please use strict-type checking
```suggestion
            if ($flag === $name) {
```